cache_test.go

```diff
@@ -524,3 +590,10 @@ func TestCacheMetricsClear(t *testing.T) {
 	c.Metrics = nil
 	c.Metrics.Clear()
 }
+
+func TestMain(m *testing.M) {
+	// Set bucketSizeSecs to 1 to avoid waiting too much during the tests.
+	bucketDurationSecs = 1
+	m.Run()
+	os.Exit(0)
```

Comment:
You are supposed to exit with the return value of m.Run, so that the exit status of the test binary reflects whether tests have passed or failed.

But the whole TestMain function can probably be replaced with a `func init() { bucketDurationSecs = 1 }` in cache_test.go